File: proc_fizzbuzz.rb
n = -> x { x }
f = -> x { x % 3 == 0 ? "fizz" : n[x] }
b = -> x { x % 5 == 0 ? "buzz" : f[x] }
fb = -> x { x % 15 == 0 ? "fizzbuzz" : b[x] }

(1..100).each { |n| p fb[n] }

File: proc_fizzbuzz.php
<?php
// fizzbuzz with an anonymous function

$fb = function ($x) {
  return $x % 15 == 0 ? "fizzbuzz" : $x % 5 == 0 ? "buzz" : $x % 3 == 0 ? "fizz" : $x;
};

for ($i = 1; $i <= 100; $i++) {
  echo $fb($i)."\n";
}
